### PR TITLE
add retcode

### PR DESCRIPTION
**Pull request by tiantn**
Signed-off-by: tian na <tiantn@cn.ibm.com>

### CODE REVIEW

File: src/__test__/testUtils.ts [hunk 1]
```diff
@@ -86,6 +86,10 @@ export async function submitHelloJob(client: ZosAccessor) {
         status = await client.getJobStatus(queryStatus);
         await sleep(5000);
     }
+    while (status.retcode) {
```
Comment (std4lqi) [inline src/__test__/testUtils.ts:89]: It's unnecessary, since it always returns `rc`.

File: src/interfaces/Job.ts [hunk 2]
```diff
@@ -57,6 +57,8 @@ export interface JobStatus extends Job {
      */
     rc?: string | number;
 
+    retcode?: string;
```
Comment (std4lqi) [inline src/interfaces/Job.ts:60]: Doc is missing.

File: src/zosAccessor.ts [hunk 3]
```diff
@@ -723,7 +723,7 @@ class ZosAccessor {
                             const deferred = Q.defer<JobStatusResult>();
                             this.getJobStatus({ jobId: queryOption.jobId, owner: option.owner })
                                 .then((jobStatus: JobStatus) => {
-                                    if (jobStatus.rc === 0) {
+                                    if (jobStatus.rc === 0 || jobStatus.retcode === 'RC 0000') {
```
Comment (std4lqi) [inline src/zosAccessor.ts:726]: We reply on rc only.

File: src/zosAccessor.ts [hunk 4]
```diff
@@ -750,17 +750,32 @@ class ZosAccessor {
...
+                    const posn2 = line.indexOf(EYE_CATCHER_2);
+                    const posn3 = line.indexOf(EYE_CATCHER_3);
+                   if (posn1 !== -1) {
+                        rc = parseInt(line.substring(posn1 + EYE_CATCHER_1.length).trim(), 10);
+                        retcode = 'RC ' + line.substring(posn1 + EYE_CATCHER_1.length).trim();
+                        }
+                    else if(posn2 !== -1) {
+                        retcode = 'ABEND ' + line.substring(posn2 + EYE_CATCHER_2.length).trim();
+                        rc = retcode;
+                        }
```
Comment (std4lqi) [inline src/zosAccessor.ts:770]: As we talked, we don't mean to have the exact same error string as z/OSMF. So no need to parse the string from my perspective.

Comment (tiantn) [reply src/zosAccessor.ts:770]: We have another issue: can't return rc when job canceled. This part is for resolving the issue. The canceled job will have ABEND S222 code in JESMSG. The next part is for SEC ERROR, the reason is similar to canceled job.

**Pull request merged**